contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,438 @@
+	Spec              CStorPoolClusterSpec   `json:"spec"`
+	Status            CStorPoolClusterStatus `json:"status"`
+}
+
+// CStorPoolClusterSpec is the spec for a CStorPoolClusterSpec resource
+type CStorPoolClusterSpec struct {
+
+
+    // MaxPools is the desired pool count in case of auto provisioning.
+    MaxPools *int        
+               `json:"maxPools"`
+    // diskType is the disk type -- e.g. sparse or disk
+    // This field is required but is not honoured in manual provisioning.
+    diskType     string                     `json:"diskType"`
+
+    // cache file is used put pool configuration which can make the import faster.
+    // This field is optional.
+    CacheFile        string   `json:"cacheFile"`    
+ 
+    // type is the raid group Possible values : stripe, mirror, raidz or raidz2.
+    
+    // stripe: It is a raid group which divides data into blocks and spreads the 
+    // data blocks across multiple block devices.
+
+    // mirror: It is a raid group which creates redundancy of data across block devices. 
+     
+    // This field is required.
+    
+    Type         PoolType `json:"poolType"`    
+ 
+    // overProvisioning if set to true allows over provisioning else does not.
+    // This field is optional i.e. by default it will be false.
+    OverProvisioning bool     `json:"overProvisioning"` //true or false
+
+    // Nodes is the list of node spec which contains details for pool
+    // provisioning on the nodes.
+    Nodes    []CStorPoolClusterNodeSpec `json:"nodes"`
+
+}
+
+// CStorPoolClusterNodeSpec is the spec for node where pool should be created.
+type CStorPoolClusterNodeSpec struct {
+
+        // Name is the name of the node.
+	Name string `json:"name"`
+
+	// BlockDeviceGroups contains the list of BD groups that should be used for
+        // pool provisioning on that node.
+	BlockDeviceGroups []CStorPoolClusterBlockDevieGroups `json:"groups"`
+}
+
+// CStorPoolClusterBlockDeviceGroups contains details of a block device group.
+type CStorPoolClusterBlockDeviceGroups struct {
+
+	// GroupName is the name of the block device group.
+	GroupName string `json:"groupName"`
+
+	// BlockDevices is the block devices present in the group.
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevice"`
+}
+
+
+// CStorPoolClusterBlockDevice contains the name details of a block device.
+type CStorPoolClusterBlockDevice struct {
+	// blockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// ID is a unique id associated to this block device either by user or system.
+	// This ID does not change even if the block device is replaced so one can
+        // think it as a slot for this block device.
+	ID string `json:"id"`
+
+        // Capacity is the size of the block device. It is populated by the system.
+        Capacity string `json:"capacity"`
+}
+
+// CStorPoolClusterStatus is for handling status of pool.
+type CStorPoolClusterStatus struct {
```

Comment:
What will be possible states of the cluster spec? Can a single field capture every possible state?